jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -790,36 +888,33 @@ else if (Violation.CASE_INSENSITIVE_METHOD.isAllowedBy(_complianceMode))
                     break;
 
                 case URI:
-                    assert _requestHandler != null;
+                    assert _requestParser;
                     int position = buffer.position();
                     int remaining = buffer.remaining();
                     switch (t.getType())
                     {
                         case SPACE:
-                            if (remaining >= (HTTP_VERSION_LEN + 1) && buffer.getInt(position) == HTTP_AS_INT)
+                            int endOfVersion = position + BYTES_IN_LONG;
+                            if (remaining >= (BYTES_IN_LONG + 2) &&
+                                buffer.get(endOfVersion) == CARRIAGE_RETURN &&
+                                buffer.get(endOfVersion + 1) == LINE_FEED)
```

Comment:
You could do a `buffer.getShort(endOfVersion) == CR_LF_SHORT` instread of two gets of a single byte.